modcc/symge.cpp

```diff
@@ -123,12 +125,14 @@ void gj_reduce(sym_matrix& A, symbol_table& table) {
         for (unsigned i = 0; i<A.nrow(); ++i) {
             if (i==p.row || A[i].index(p.col)==msparse::row_npos) continue;
             A[i] = row_reduce(p.col, A[i], A[p.row], define_sym);
+            row_partition.push_back(row_partition.back() + A[i].size());
         }
 
         if (remaining_rows.empty()) {
             break;
         }
     }
+    return row_partition;
```

Comment:
`row_partition` content seems to depend rather crucially on the details of how row `row_reduce` makes new symbols, and so it feels a bit brittle. Is it possible for the normalization code to just walk the rows of the matrix `A_` itself? I think it will make the code both clearer and more robust.

Reply:
Yes, I agree. It would be clearer.